Sparkle/SUFileManager.m

```diff
@@ -13,11 +13,7 @@
 #include <sys/errno.h>
 #include <sys/time.h>
 
-#pragma clang diagnostic push
-// This is a private constant but it's referenced from our unit tests
-#pragma clang diagnostic ignored "-Wmissing-variable-declarations"
-NSString *SUAppleQuarantineIdentifier = @"com.apple.quarantine";
-#pragma clang diagnostic pop
+static char SUAppleQuarantineIdentifier[] = "com.apple.quarantine";
```

Comment:
No, I think that warning happens only because of [a design mistake in C's type system](http://c-faq.com/ansi/constmismatch.html). 

In C `const char *` argument can accept both `char *` and `const char *` values. However, an argument with `const char **` type only accepts `const char **` and not `char **`.
 
Therefore, in practice, you can't have a C function that takes an array of pointers and guarantees it won't modify them, because it will be unusable with non-const values.

So Apple couldn't declare that this function doesn't modify the arguments even if they wanted to.

So I'm relying on the fact that:

* Documentation doesn't say anything about them being modified
* The type is `char *const *` which is the closest practical think you can do to declare an immutable array of pointers